### DISCUSSION TITLE
Making player 2 default player for solo mode

### DISCUSSION THREAD

**Discussion by veckin** [FunkinCrew/Funkin#1576] [Q&A]
For a specific level, I'm trying to get it so that player 2 is the sole playable character, game over activation and all. Like, their side is the one used in story mode and freeplay.

I tried moving the positions of the characters and flipping them in the code, but that made it more confusing since the player beatmaps would still be played on the right side. Is there a simpler way to make the left the 'boyfriend' side, or did I overlook something?

**Comment by cuckydev** [comment]
Not the correct place to ask.

**Comment by ghost** [comment]
> Not the correct place to ask.

kinda is

ALSO, HELLO PERSON WHO PORTED TO PS1

**Comment by veckin** [comment]
Probably should elaborate here: Using the dev mode to switch the player positions only resulted in the camera following BF slightly out of boundaries (and floating), and while the dev mode also allows one to flip the beatmaps, the right side (and character attached to it) is still the one necessary to play. 

When accounting for this in the code,  I had two temporary solutions:

Solution 1 was to try and create an instance of P2 which counted as a "Boyfriend" class, so that I could activate a Game Over using a conditional. The result was P2 starting off in P1's direction, and having to flipX on it to have him appear as it normally does. Long story short, the conditional didn't work and it just led to the Game Over screen shifting to P1's position as usual.
```
case 'philly': //don't know if this part will be relevant
				dad = new Boyfriend(dad.x, dad.y, SONG.player2);
				dad.flipX = true;
				gf.y -=550;
				dad.x +=300;
				dad.y += 300;

if (dad.curCharacter.contains('bf') && Std.isOfType(dad, Boyfriend))
				{
					openSubState(new GameOverSubstate(dad.getScreenPosition().x, dad.getScreenPosition().y));
				}
			else 
				{
					openSubState(new GameOverSubstate(boyfriend.getScreenPosition().x, boyfriend.getScreenPosition().y));
				}
```


Solution 2 was to just move P1 and P2's x positions to the opposite side and flip them in the hard-coding of that stage. When P1 is moved it does at least have game over move with it, but the Game Over sprite isn't flipped along with it and it still doesn't change the fact that the health bar and beatmaps still shows P1 on the right and P2 on the left. In addition, P2 doesn't move all that much when adjusting its x position. It will move a good bit, but it's not like it's going to the other half of the screen.

Now I know there's something I'm missing here, since most of this is cosmetic changes. But I'm not too familiar with how to really change things around, so I'd appreciate some feedback.
